coreos-kubernetes-multimaster/azuredeploy.json

```diff
@@ -0,0 +1,832 @@
n        Description=Kubernetes Kubelet Server\n        Documentation=https://github.com/kubernetes/kubernetes\n\n        [Service]\n        ExecStartPre=/bin/mkdir -p /var/lib/kubelet\n        ExecStartPre=/bin/wget -O /opt/kubernetes/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/',parameters('hyperkubeVersion'),'/bin/linux/amd64/kubectl\n        ExecStartPre=/bin/chmod 755 /opt/kubernetes/bin/kubectl\n        ExecStartPre=/bin/mount --bind /var/lib/kubelet /var/lib/kubelet\n        ExecStartPre=/bin/mount --make-shared /var/lib/kubelet\n        ExecStart=/usr/bin/docker run \\\n          --net=host \\\n          --pid=host \\\n          --privileged \\\n          -v /dev:/dev \\\n          -v /sys:/sys:ro \\\n          -v /var/run:/var/run:rw \\\n          -v /var/lib/docker/:/var/lib/docker:rw \\\n          -v /var/lib/kubelet/:/var/lib/kubelet:shared \\\n          -v /etc/kubernetes:/etc/kubernetes:ro \\\n          gcr.io/google_containers/hyperkube-amd64:',parameters('hyperkubeVersion'),' \\\n          /hyperkube kubelet \\\n            --cloud-provider=azure \\\n            --cloud-config=/etc/kubernetes/conf/azure.json \\\n            --address=0.0.0.0 \\\n            --allow-privileged=true \\\n            --enable-server \\\n            --enable-debugging-handlers \\\n            --kubeconfig=/etc/kubernetes/conf/kubeconfig.yaml \\\n            --config=/etc/kubernetes/manifests \\\n            --cluster-dns=10.240.0.10 \\\n            --cluster-domain=cluster.local \\\n            --v=2 \\\n            --api-servers=http://localhost:8080 \\\n            --register-schedulable=false \\\n            --pod-cidr=10.244.0.0/16\n        Restart=always\n        RestartSec=5\n\n        [Install]\n        WantedBy=multi-user.target\n    - name: \"kube-addons.service\"\n      command: \"start\"\n      content: |\n        [Unit]\n        Description=Install kubernetes addons\n        After=kubelet-master.service\n        [Service]\n        ExecStartPre=/opt/kubernetes/bin/wupiao 127.0.0.1 8080\n        ExecStartPre=/bin/bash -c \"while [ $(curl -s -L http://127.0.0.1:8080/healthz) != ok ]; do echo Waiting for cluster readiness ; sleep 5; done\"\n        ExecStart=/opt/kubernetes/bin/kubectl apply -R -f /etc/kubernetes/addons/\n        ExecStart=/opt/kubernetes/bin/kubectl apply -f https://raw.githubusercontent.com/kelseyhightower/kubernetes-the-hard-way/master/deployments/kubedns.yaml\n        ExecStart=/opt/kubernetes/bin/kubectl apply -f https://raw.githubusercontent.com/kubernetes/kubernetes/master/cluster/addons/dashboard/dashboard-controller.yaml\n        ExecStart=/opt/kubernetes/bin/kubectl apply -f https://raw.githubusercontent.com/kubernetes/kubernetes/master/cluster/addons/dashboard/dashboard-service.yaml\n        ExecStart=/opt/kubernetes/bin/kubectl apply -f https://raw.githubusercontent.com/kubernetes/kubernetes/master/cluster/addons/node-problem-detector/node-problem-detector.yaml\n        Type=oneshot\n'))]",
```

Comment:
are you sure you want to be passing the client secret here? As I understand it, customData is not a secure way to pass secrets.